src/main/java/com/google/sps/youtube/YoutubeResponse.java

```diff
@@ -0,0 +1,9 @@
+
+package com.google.sps.youtube;
+
+import java.io.IOException;
+import com.google.api.services.youtube.model.ChannelListResponse;
+
+public interface YoutubeResponse{
```

Comment:
So I assume you defined this interface to represent your service so you can fake/mock it out?   First I don't think it's necessary because you can just define a mock on the YouTube api client since it's not final (unless you ran into some issue?):

https://developers.google.com/resources/api-libraries/documentation/youtube/v3/java/latest/com/google/api/services/youtube/YouTube.html

So you don't actually need the interface.

But let's say it were final, then you would probably want the interface, but you'd want to name it YouTubeClient or something (the class name implies that it itself is a response, but it has methods that return a response).  It itself isn't a response; it has methods that return a response.

From an OOP perspective, you wouldn't want to have YoutubeInfoScraper implements YouTubeClient (after my rename suggestion) because then you're implying that the Scraper is a YouTubeClient, but it isn't. (See Effective java, prefer composition over inheritance.). It *depends* on a YouTubeClient in which case you'd use DI and inject a fake vs. real YouTubeClient and assign that to a field (which it looks like you're already doing).

Reply:
Removed the interface.